core/src/main/java/org/apache/carbondata/core/indexstore/BlockletDataMapIndexStore.java

```diff
@@ -151,10 +153,10 @@ private BlockMetaInfo createBlockMetaInfo(String carbonDataFile) throws IOExcept
       }
       if (missedIdentifiers.size() > 0) {
         SegmentIndexFileStore indexFileStore = new SegmentIndexFileStore();
-
+        List<String> filesRead = new ArrayList<>();
```

Comment:
use Set in place of List...Set will have a better performance as compared to list as your data grows